web/js/codeworld.js

```diff
@@ -221,6 +221,59 @@ function initCodeworld() {
         }
     });
     window.codeworldEditor.refresh();
+    window.codeworldEditor.on("cursorActivity", function() {
+        const prevDiv = document.getElementById("function-details");
+        if (prevDiv) prevDiv.remove();
+
+        const enclosingFunction = getEnclosingFunction(window.codeworldEditor);
+
+        if (!enclosingFunction || !enclosingFunction.string ||  !window.codeWorldSymbols[enclosingFunction.string]) {
+            return;
+        }
+        const cursor = window.codeworldEditor.getCursor();
+        const currentToken = window.codeworldEditor.getTokenAt(cursor);
+        const lineTokens = window.codeworldEditor.getLineTokens(cursor.line);
+        const argIndex =
+            lineTokens.filter(
+                elem =>
+                    getEnclosingFunction(
+                        window.codeworldEditor,
+                        {
+                            line: cursor.line, ch: elem.end
+                        })
+                        .string === enclosingFunction.string
+                    && elem.type && elem.type !== "keyword"
+                    && elem.type!=="bracket-0"
+                    && elem.type !== "built-in"
+            ).findIndex(elem => (elem.start === currentToken.start || elem.end === currentToken.start || elem.start === currentToken.end || elem.end === currentToken.end));
```

Comment:
Are you sure that counting tokens works here?  There are lots of things, like passing an expression as an argument, that  would change the number of tokens.  If the arguments are in a tuple, I'd expect you to look for commas in particular.

Reply:
Got it. I think I can solve all of the problems by doing something like
```javascript
line = line.substr(0, token.end);
let level = 0;
let i = line.length - 1;
let argIndex = 0;
for (; i>=0; i--){
if (line[i] === ')') level++;
if (line[i] === '(' && level === 0) break;  
if (line[i] === '(' && level > 0) level--;  
if (level > 0) continue;
if (line[i] === ',') argIndex++;  // (no. of commas in current level = argIndex)
}
const enclosing function = // logic to get token preceding index and see if it's declaration is known.
```